File: Frontend/src/app/models/Investments/Investment.ts

import { Base } from "../Base.js";
import { Frequency } from "../Admin/frequency.js";
import { User } from "../Admin/user.js";
import { UserPaymentMethod } from "../Admin/UserPaymentMethod.js";
import { InvestmentCategory } from "../Admin/InvestmentCategory.js";
import { InvestmentType } from "../Admin/InvestmentType.js";

export class Investment extends Base 
{   
    public user? : User;
    public frequency? : Frequency;
    public invested? : boolean;
    public value? : number;
    public investedTo? : string;
    public investedVia? : UserPaymentMethod;
    public description?: string;
    public investmentType?: InvestmentType;
    public investmentCategory?: InvestmentCategory;
    public investedOn? : Date;
    public tenure?: number;
    public growthRate?: number;

}

File: Features/DL/Entities/Investment.cs
namespace Features.DL.Entities
{
    public class Investment: BaseModel
    {
        public virtual User User { get; set; }
        public virtual Frequency Frequency { get; set; }
        public virtual Boolean Invested { get; set; }

        public virtual Decimal Value { get; set; }
        public virtual string InvestedTo { get; set; }

        public virtual Decimal GrowthRate { get; set; }

        public virtual UserPaymentMethod InvestedVia { get; set; }

        public virtual InvestmentType InvestmentType { get; set; }

        public virtual InvestmentCategory InvestmentCategory { get; set; }
        public virtual string Description { get; set; }

        public virtual DateTime? InvestedOn { get; set; }

        public virtual Decimal Tenure { get; set; }

    }
}
